src/components/layout/signIn/SignIn.tsx

```diff
@@ -0,0 +1,81 @@
+import { FC, useEffect } from 'react';
+
+import { Paper, Center, Title, Button, useMantineTheme } from '@mantine/core';
+import { AuthError } from 'firebase/auth';
+import { FcGoogle } from 'react-icons/fc';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+
+import { checkUserInDB, signInWithPopupApi } from 'api/api';
+import { color } from 'components/layout/signIn/config';
+import FormSignIn from 'components/shared/formSignIn/FormSignIn';
+import {
+  fetchUserRequest,
+  fetchUserSuccess,
+  fetchUserFailure,
+  setUserIsLoggedIn,
+} from 'store/user/userData.api';
+
+const SignIn: FC = () => {
+  const theme = useMantineTheme();
+  const dispatch = useDispatch();
+  const { user, token, isLoggedIn, error } = useSelector((state: any) => state.user);
```

Comment:
do not use [useSelector destructuring](https://stackoverflow.com/questions/59072200/useselector-destructuring-vs-multiple-calls) and do not use `any`